framework/src/modules/random/module.ts

```diff
@@ -243,16 +238,28 @@ export class RandomModule extends BaseModule {
 				count: 0,
 			};
 		}
-		// If checkpoint is reached then increment the nextCheckpointIndex taking integer into account
-		const nextCheckpointIndex =
-			nextCount % hashOnion.distance === 0
-				? Math.ceil(nextCount / hashOnion.distance) + 1
-				: Math.ceil(nextCount / hashOnion.distance);
+
+		// if at the checkpoint, return the hash available from the store
+		const distanceAfterCheckpoint = nextCount % hashOnion.distance;
+		if (!distanceAfterCheckpoint) {
```

Comment:
Functionally, that is the same thing. No distance means that distance is 0.
 
But OK, let's stick with numbers. Changed the condition to check if it is zero instead of falsy.

Reply:
Also, I think it's missing the test case where it was crashing